src/lib/Output/Generator/Json.php

```diff
@@ -249,79 +69,59 @@ public function startList($name)
         $this->json = $array;
     }
 
-    /**
-     * End list.
-     *
-     * @param string $name
-     */
-    public function endList($name)
+    #[\Override]
+    public function startAttribute(string $name, mixed $value): void
     {
-        $this->checkEndList($name);
+        $this->checkStartAttribute($name);
 
-        $this->json = $this->json->getParent();
+        $this->json->{'_' . $name} = $value;
     }
 
-    /**
-     * Start attribute.
-     *
-     * @param string $name
-     * @param string $value
-     */
-    public function startAttribute($name, $value)
+    #[\Override]
+    public function getMediaType(string $name): string
     {
-        $this->checkStartAttribute($name);
-
-        $this->json->{'_' . $name} = $value;
+        return $this->generateMediaTypeWithVendor($name, 'json', $this->vendor);
     }
 
-    /**
-     * End attribute.
-     *
-     * @param string $name
-     */
-    public function endAttribute($name)
+    #[\Override]
+    public function serializeBool(mixed $boolValue): bool
     {
-        $this->checkEndAttribute($name);
+        return (bool)$boolValue;
     }
 
-    /**
-     * Get media type.
-     *
-     * @param string $name
-     *
-     * @return string
-     */
-    public function getMediaType($name)
+    #[\Override]
+    public function getData(): DataObjectInterface
     {
-        return $this->generateMediaTypeWithVendor($name, 'json', $this->vendor);
+        return $this->json;
     }
 
-    /**
-     * Generates a generic representation of the scalar, hash or list given in
-     * $hashValue into the document, using an element of $hashElementName as
-     * its parent.
-     *
-     * @param string $hashElementName
-     * @param mixed $hashValue
-     */
-    public function generateFieldTypeHash($hashElementName, $hashValue)
+    #[\Override]
+    public function getEncoderContext(array $data): array
     {
-        $this->fieldTypeHashGenerator->generateHashValue(
-            $this->json,
-            $hashElementName,
-            $hashValue
-        );
+        return [];
     }
 
     /**
-     * Serializes a boolean value.
-     *
-     * @param bool $boolValue
+     * Convert ArrayObjects to arrays.
      *
-     * @return bool
+     * Recursively convert all ArrayObjects into arrays in the full data
+     * structure.
      */
-    public function serializeBool($boolValue)
+    private function convertArrayObjects(mixed $data): mixed
     {
-        return (bool)$boolValue;
+        if ($data instanceof Json\ArrayObject) {
+            // @todo: Check if we need to convert arrays with only one single
+            // element into non-arrays /cc cba
+            $data = $data->getArrayCopy();
+            foreach ($data as $key => $value) {
+                $data[$key] = $this->convertArrayObjects($value);
+            }
+        } elseif ($data instanceof Json\JsonObject) {
+            foreach ($data as $key => $value) {
```

Comment:
> Argument of an invalid type Ibexa\Rest\Output\Generator\Json\JsonObject supplied for foreach, only iterables are supported.

Does this code work?

Reply:
It's an old code but yes, it works. It iterates over JsonObject's properties